hudi-common/src/main/java/org/apache/hudi/common/table/cdc/HoodieCDCExtractor.java

```diff
@@ -114,6 +114,24 @@ public Map<HoodieFileGroupId, List<HoodieCDCFileSplit>> extractCDCFileSplits() {
     ValidationUtils.checkState(commits != null, "Empty commits");
 
     Map<HoodieFileGroupId, List<HoodieCDCFileSplit>> fgToCommitChanges = new HashMap<>();
+
```

Comment:
hey @danny0405 @yihua : Can you folks rope in someone to assist w/ CDC fixes here. 
I am not sure if this is the right fix. 

but let me go over the gist. 
There was a test failure which I was trying to chase and realized we might have some gaps here. 

I am not going go to over exact scenario that the test failed. but the gist is this.

dc1
dc2
rc3
dc4
clean5

in this case, we are looking to parse and fetch base files for dc1 and dc2 as well which could have been cleaned up by the cleaner and may not exists. 

Before this fix, the test was set up such that, timeline was 
rc3
dc4
clean5

and so we did not hit this issue. When I made the fix to ensure getLatestDeltaCommits does the right thing (which archival is using btw), we ended up w/ timeline as 
dc1
dc2
rc3
dc4
clean5

and   private HoodieCDCFileSplit parseWriteStat(
      HoodieFileGroupId fileGroupId,
      HoodieInstant instant,
      HoodieWriteStat writeStat,
      WriteOperationType operation) {
in HoodieCDCExtractor is failing.

Reply:
So, I tried making a fix. 
first collect all replacedFileIds. and then ignore those while parsing other commit metadata. 

I was able to get past the parsing issue. but assertions for record counts are failing. probably anyone who has worked on CDC might be better to fix this. 

Test case of interest: 
TestCDCDataFrameSuite.testMORDataSourceWrite

Reply:
> we are looking to parse and fetch base files for dc1 and dc2 as well which could have been cleaned up by the cleaner and may not exists.

Are you saying the cleaning state has changed because of the archiving? If that is true, maybe we should just adjust the cleaning strategy to keep more historical commits.

Reply:
nope. I am saying HoodieCDCExtractor.java has some inherent bug which I am trying to fix here. 

lets say timeline is as follows

dc1
dc2
rc3
dc4
clean5 // cleans up data files from dc1 and dc2 since it was replaced by rc3. 

as per master, HoodieCDCExtractor goes over commit metadata in activetimeline and tries to deduce base files for log files found. In this case, all data files from dc1 and dc2 are already deleted by clean5. And so we might hit file not found issue as per master.

Reply:
> In this case, all data files from dc1 and dc2 are already deleted by clean5. but HoodieCDCExtractor tries to parse data files from dc1 and dc2

That is by design, currently only consuming from alive files is supported.